base/loading.jl

```diff
@@ -2700,7 +2700,7 @@ end
 
 function get_preferences(uuid::Union{UUID,Nothing} = nothing)
     merged_prefs = Dict{String,Any}()
-    for env in reverse(load_path())
+    for env in reverse!(load_path())
```

Comment:
This seems potentially problematic. `load_path` can return a reference to a cached array